code/modules/roguetown/roguejobs/artificer/contraptions.dm

```diff
@@ -0,0 +1,235 @@
+		shake_camera(user, 1, 1)
+		var/datum/effect_system/spark_spread/S = new()
+		var/turf/front = get_turf(I)
+		S.set_up(1, 1, front)
+		S.start()
+		var/probability = rand(1, 100)
+		if(!current_charge)
+			misfire(I, user)
+			return
+		if(probability <= 5)
+			misfire(I, user)
+		else if(probability <= 40)
+			if(current_charge < charge_per_source)
+				current_charge += 1
+			misfire_chance = rand(1, 30)
+		else
+			misfire_chance = rand(10, 100)
+	..()
+
+/obj/item/contraption/proc/play_clock_sound()
+	playsound(src, 'sound/misc/clockloop.ogg', 25, TRUE)
+
+/obj/item/contraption/attack_obj(obj/O, mob/living/user)
+	if(!current_charge)
+		flick(off_icon, src)
+		to_chat(user, span_info("The contraption beeps! It requires \a [initial(accepted_power_source.name)]!"))
+		playsound(src, 'sound/magic/magic_nulled.ogg', 100, TRUE)
+		return
+
+/obj/item/contraption/wood_metalizer
+	name = "wood metalizer"
+	desc = "A creation of genious or insanity. This cursed contraption is somehow able to turn wood into metal."
+	icon_state = "metalizer"
+	on_icon = "metalizer_flick"
+	off_icon = "metalizer_off"
+	w_class = WEIGHT_CLASS_BULKY
+	smeltresult = /obj/item/ingot/bronze
+	misfire_chance = 15
+	charge_per_source = 5
+
+/obj
+	/// This is the result when the wood metalizer artifact is used on this item
+	var/metalizer_result
+	/// The smelting result, used by the smelter or by the portable smelter
+	var/smeltresult
+
+/obj/item/contraption/wood_metalizer/attack_obj(obj/O, mob/living/user)
+	..()
+	if(!current_charge)
+		return
+	if(!O.metalizer_result)
+		to_chat(user, span_info("The [name] refuses to function."))
+		playsound(user, 'sound/items/flint.ogg', 100, FALSE)
+		flick(off_icon, src)
+		var/datum/effect_system/spark_spread/S = new()
+		var/turf/front = get_turf(O)
+		S.set_up(1, 1, front)
+		S.start()
+		return
+	else
+		var/skill = user.mind.get_skill_level(/datum/skill/craft/engineering)
+		if(istype(O, /obj/structure/mineral_door/wood)) //This is to ensure the new door will retain its lock
+			var/obj/structure/mineral_door/wood/I = O
+			var/obj/structure/mineral_door/wood/new_door = new I.metalizer_result(get_turf(I))
+			new_door.locked = I.locked
+			if(I.lockid)
+				new_door.lockid = I.lockid
+			qdel(I)
+		else
+			var/obj/I = O
+			new I.metalizer_result(get_turf(I))
+			qdel(I)
+		flick(on_icon, src)
+		current_charge -= 1
+		shake_camera(user, 1, 1)
+		playsound(src, 'sound/magic/swap.ogg', 100, TRUE)
+		user.mind.add_sleep_experience(/datum/skill/craft/engineering, (user.STAINT / 2))
+		if(misfire_chance && prob(max(0, misfire_chance - user.goodluck(2) - skill)))
+			misfire(O, user)
+		if(!current_charge)
+			addtimer(CALLBACK(src, PROC_REF(battery_collapse), O, user), rand(5))
+		return
```

Comment:
the first ELSE and subsequent indentation is redundant

Reply:
Okay ball buster 😡